### ISSUE TITLE
inception_v3 decode_predictions incorrect labels

### ISSUE THREAD

**Issue by bdwyer2** [fchollet/keras#7822]
It seems like the labels return by `decode_predictions` may be incorrect:

```python
import keras
from keras.preprocessing import image
from keras.applications.inception_v3 import decode_predictions
import numpy as np

model=keras.applications.inception_v3.InceptionV3()

print('car')
car_img=image.load_img('car.jpg', target_size=(299, 299))
car_img=image.img_to_array(car_img)
car_img=np.expand_dims(car_img, axis=0)
car_pred=model.predict(car_img)
print(decode_predictions(car_pred, top=5))

print('cat')
cat_img=image.load_img('cat.jpg', target_size=(299, 299))
cat_img=image.img_to_array(cat_img)
cat_img=np.expand_dims(cat_img, axis=0)
cat_pred=model.predict(cat_img)
print(decode_predictions(cat_pred, top=5))
```

returns:
```
car
[[('n03047690', 'clog', 1.0), ('n06359193', 'web_site', 2.0100744e-11), ('n01924916', 'flatworm', 1.2693379e-14), ('n03950228', 'pitcher', 2.014e-15), ('n04131690', 'saltshaker', 9.2570057e-21)]]
cat
[[('n03047690', 'clog', 1.0), ('n06359193', 'web_site', 1.2098395e-36), ('n09468604', 'valley', 6.4327662e-37), ('n03950228', 'pitcher', 8.219755e-38), ('n02342885', 'hamster', 0.0)]]
```

Am I doing something wrong here? Both of these images return the correct label when passed to `classify_image.py` in TensorFlow

![cat](https://user-images.githubusercontent.com/19673623/30083486-6d47cd3c-9243-11e7-9549-b1ccf8d6a41a.jpg)
![car](https://user-images.githubusercontent.com/19673623/30083487-6d5d3c62-9243-11e7-8022-104ccf03b33e.jpg)

**Comment by myutwo150** [comment]
Use `keras.applications.inception_v3.preprocess_input()` before prediction.

**Comment by bdwyer2** [comment]
@myutwo150 perfect, thanks!
